File: codility/lessons/4_counting_elements/perm_check.cpp
// https://app.codility.com/programmers/lessons/4-counting_elements/perm_check/

#include <unordered_set>
#include <vector>
using namespace std;

int solution(vector<int> &A) {
    auto values = unordered_set<int>(A.begin(), A.end());
    int min_allowed_value = 1,
        max_allowed_value = (int)A.size();
    for (auto value : values) {
        if (value < min_allowed_value || value > max_allowed_value) {
            return 0;
        }
    }
    return (int)(values.size() == A.size());
}

File: codility/lessons/4_counting_elements/perm_check.py
# https://app.codility.com/programmers/lessons/4-counting_elements/perm_check/

from collections import Counter

def solution(A):
    occurences = Counter(A)
    start, end = 1, len(A) + 1
    for number in range(start, end):
        count = occurences.get(number)
        if count != 1:
            return 0
    return 1
